### PR TITLE
Swift conversion review: vineet

### PR DESCRIPTION
**Pull request by davidme-stripe**
These are Swift conversion files for vineet to review. None of them were converted by vineet.

### CODE REVIEW

File: Stripe/STPTheme.swift [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+import UIKit
+
+/// STPTheme objects can be used to visually style Stripe-provided UI. See https://stripe.com/docs/mobile/ios/standard#theming for more information.
+public class STPTheme: NSObject {
+
+  /// The default theme used by all Stripe UI. All themable UI classes, such as `STPAddCardViewController`, have one initializer that takes a `theme` and one that does not. If you use the one that does not, the default theme will be used to customize that view controller's appearance.
+  @objc public static let defaultTheme = STPTheme()
+
+  /// The primary background color of the theme. This will be used as the `backgroundColor` for any views with this theme.
+  @objc public var primaryBackgroundColor: UIColor = STPThemeDefaultPrimaryBackgroundColor
```
Comment (vineet-stripe) [inline Stripe/STPTheme.swift:18]: I guess we're ditching the `null_resettable` behavior for this property? Is there a new approach to resetting, if someone was relying on that behavior?

**Review by vineet-stripe — APPROVED** [review]
Looks good! Mostly style nits

File: Stripe/STPTelemetryClient.swift [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    }
+    request?.httpBody = data
+    var task: URLSessionDataTask?
+    if let request = request {
+      task = urlSession?.dataTask(with: request as URLRequest)
+    }
+    task?.resume()
+  }
+
+  private var urlSession: URLSession?
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPTelemetryClient.swift:62]: Fixed (and cleaned this up a bit more, as most of the initializers weren't used)

File: Stripe/STPTelemetryClient.swift [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    if sharedInstanceSharedClient != nil {
+      return sharedInstanceSharedClient as! Self
+    } else {
+      sharedInstanceSharedClient = STPTelemetryClient.init(sessionConfiguration: config)
+      return sharedInstanceSharedClient as! Self
+    }
+  }
+
+  func addTelemetryFields(toParams params: inout [String: Any]) {
+    params["muid"] = muid() ?? ""
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPTelemetryClient.swift:26]: This shouldn't even be optional, tbh

File: Stripe/STPTelemetryClient.swift [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+class STPTelemetryClient: NSObject {
+  static var sharedInstanceSharedClient: STPTelemetryClient?
+
+  @discardableResult @objc class func sharedInstance() -> Self {
+    let config = STPAPIClient.sharedUrlSessionConfiguration
+    if sharedInstanceSharedClient != nil {
+      return sharedInstanceSharedClient as! Self
+    } else {
+      sharedInstanceSharedClient = STPTelemetryClient.init(sessionConfiguration: config)
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPTelemetryClient.swift:20]: I prefer without `init`, I think it's just an Objective-C-ism that we carried over. Maybe it's worth a pass to switch everywhere.

File: Stripe/STPTelemetryClient.swift [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class STPTelemetryClient: NSObject {
+  static var sharedInstanceSharedClient: STPTelemetryClient?
+
+  @discardableResult @objc class func sharedInstance() -> Self {
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPTelemetryClient.swift:15]: Agreed, this wasn't really cleaned up after being autoconverted.

File: Stripe/STPTheme.swift [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+//  Stripe
+//
+//  Created by Jack Flintermann on 5/3/16.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+import UIKit
+
+/// STPTheme objects can be used to visually style Stripe-provided UI. See https://stripe.com/docs/mobile/ios/standard#theming for more information.
+public class STPTheme: NSObject {
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPTheme.swift:12]: We should probably do this everywhere too, though our old SDK was always subclassable.

File: Stripe/STPThreeDSButtonCustomization.swift [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+  /// The background color of the button.
+  /// The default for .resend and .cancel is clear.
+  /// The default for .submit, .continue, and .next is blue.
+
+  @objc public var backgroundColor: UIColor {
+    get {
+      return buttonCustomization.backgroundColor
+    }
+    set(backgroundColor) {
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPThreeDSButtonCustomization.swift:72]: ++, changing it

File: Stripe/STPThreeDSCustomizationSettings.swift [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import Foundation
+import Stripe3DS2
+import UIKit
+
+/// `STPThreeDSCustomizationSettings` provides customization options for 3DS2 authentication flows in your app.
+public class STPThreeDSCustomizationSettings: NSObject {
+  /// Returns an `STPThreeDSCustomizationSettings` preconfigured with the default
+  /// Stripe UI settings and a 10 minute `authenticationTimeout`.
+  @objc
+  public class func defaultSettings() -> STPThreeDSCustomizationSettings {
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPThreeDSCustomizationSettings.swift:18]: This API doesn't make much sense to me: `default` is a weird name for this in Swift, and init() has the same result. I'm just going to deprecate it.

File: Stripe/STPTheme.swift (hunk 1 above)
Comment (davidme-stripe) [reply Stripe/STPTheme.swift:18]: Yeah, we decided to make these nonnull instead, as we couldn't think of a reason someone would want to use them in a null_resettable way. If someone wants to reset to the default value, they can create an STPTheme() and get the initial values from there.

File: Stripe/STPTelemetryClient.swift [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  func sendTelemetryData() {
+    if !STPTelemetryClient.shouldSendTelemetry() {
+      return
+    }
+    let path = "ios-sdk-1"
+    let url = URL(string: "https://m.stripe.com")?.appendingPathComponent(path)
+    var request: NSMutableURLRequest?
+    if let url = url {
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPTelemetryClient.swift:42]: URL should never be nil, I'll rewrite this

File: Stripe/STPTelemetryClient.swift [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+    }
+    request?.httpMethod = "POST"
+    request?.setValue("application/json", forHTTPHeaderField: "Content-Type")
+    let payload = self.payload()
+    var data: Data?
+    do {
+      data = try JSONSerialization.data(
+        withJSONObject: payload, options: JSONSerialization.WritingOptions(rawValue: 0))
+    } catch {
+    }
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPTelemetryClient.swift:53]: We don't have a good story for logging errors in the SDK, though we could have an assert. I'll just end the function here for now, there's no reason to send a request with an empty body. We're going to write a new telemetry class ~this year.

File: Stripe/STPTelemetryClient.swift [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  init(sessionConfiguration config: URLSessionConfiguration?) {
+    super.init()
+    if let config = config {
+      urlSession = URLSession(configuration: config)
+    }
+  }
+
+  func muid() -> String? {
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/STPTelemetryClient.swift:83]: Yeah, all of these surrounding functions should be non-optional vars

File: Stripe/STPTelemetryClient.swift [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  var muid : String {
+    get {
+      let muid = UIDevice.current.identifierForVendor?.uuidString
+      return muid ?? ""
+    }
+  }
+
+  var language = Locale.autoupdatingCurrent.identifier
```
Comment (vineet-stripe) [inline Stripe/STPTelemetryClient.swift:69]: Are these properties supposed to be `private`? They aren't exposed in `STPTelemetryClient.h`

Comment (davidme-stripe) [reply Stripe/STPTelemetryClient.swift:69]: They should be, but in the interest of time we haven't really been enforcing internal vs private boundaries. I'll fix this file up, at least.

**Pull request merged**